client-common/src/main/java/org/apache/cassandra/sidecar/common/response/LifecycleInfoResponse.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+package org.apache.cassandra.sidecar.common.response;
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.apache.cassandra.sidecar.common.data.LifecycleCassandraState;
+import org.apache.cassandra.sidecar.common.data.LifecycleStatus;
+
+/**
+ * A class representing a response for the {@code LifecycleInfoRequest}.
+ */
+public class LifecycleInfoResponse
+{
+
+    private final LifecycleCassandraState currentState;
+    private final LifecycleCassandraState desiredState;
+    private final LifecycleStatus status;
+    private final String lastUpdate;
+
+    /**
+     * Constructs a {@link LifecycleInfoResponse} object with the {@code currentState}, {@code intendedState}, {@code result},
+     * and {@code message}
+     *
+     * @param currentState the current state of the Cassandra node
+     * @param desiredState the intended state of the Cassandra node
+     * @param status the result of the last lifecycle operation
+     * @param lastUpdate a message providing additional context about the last lifecycle operation
+     */
+    @JsonCreator
+    public LifecycleInfoResponse(@JsonProperty("current_state") LifecycleCassandraState currentState,
+                                 @JsonProperty("desired_state") LifecycleCassandraState desiredState,
+                                 @JsonProperty("status") LifecycleStatus status,
+                                 @JsonProperty("last_update") String lastUpdate)
+    {
+        this.currentState = Objects.requireNonNull(currentState, "State must be non-null");
+        this.desiredState = desiredState;
+        this.status = status;
+        this.lastUpdate = lastUpdate;
+    }
+
+    /**
+     * @return the current state of the Cassandra node
+     */
+    @JsonProperty("current_state")
+    public LifecycleCassandraState currentState()
+    {
+        return currentState;
+    }
+
+    /**
+     * @return the intended state of the Cassandra node
+     */
+    @JsonProperty("desired_state")
+    public LifecycleCassandraState desiredState()
+    {
+        return desiredState;
+    }
+
+    /**
+     * @return the status of the last lifecycle state
+     */
+    @JsonProperty("status")
+    public LifecycleStatus status()
+    {
+        return status;
+    }
+
+    /**
+     * @return message providing additional context about the last lifecycle operation
+     */
+    @JsonProperty("last_update")
+    public String lastUpdate()
+    {
+        return lastUpdate;
+    }
+
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (!(o instanceof LifecycleInfoResponse)) return false;
+        LifecycleInfoResponse that = (LifecycleInfoResponse) o;
+        return currentState == that.currentState && desiredState == that.desiredState && status == that.status && Objects.equals(lastUpdate, that.lastUpdate);
```

Comment:
NIT for better readability 
```suggestion
        return currentState == that.currentState
               && desiredState == that.desiredState
               && status == that.status
               && Objects.equals(lastUpdate, that.lastUpdate);
```